cmd/snap/main.go

```diff
@@ -130,11 +130,16 @@ func init() {
 	}
 }
 
+var missingLoginHelp = "Access denied (snap login --help).\n"
+
 func main() {
 	if err := run(); err != nil {
 		if e, ok := err.(*flags.Error); ok && e.Type == flags.ErrHelp {
 			fmt.Fprintf(Stdout, "%v\n", err)
 			os.Exit(0)
+		} else if e, ok := err.(*client.Error); ok && e.StatusCode == 401 && e.Message == "Unauthorized" {
```

Comment:
@niemeyer Thanks for this suggestion. One complication here is that the 401 is returned when we try to install something without credentials. But also when we try to log into the store with an invalid password. In both cases we get a 401 and have no credentials stored in the client (yet). 

I pushed 674f3ee01c6d4fca74d30e2c40949cf91df5cb84 that adds an explicit error kind "login-required" that makes it explicit. Please let me know if that looks sensible or if you prefer a different approach to tackle this (we could also add an explicit InvalidCredentials error kind).